compiler/src/IROpt.hs

```diff
@@ -178,6 +178,8 @@ canFailOrHasEffects expr = case expr of
         Basics.IsTuple -> False
         Basics.IsList -> False
         Basics.IsRecord -> False
+        -- Boolean negation is safe
+        Basics.Not -> False
```

Comment:
Similar to `Basics.Plus` and others can fail if the argument is not a number, this can fail if the argument is not a Boolean? Or do you intend to use truthy-ness similar to JavaScript? If so, disregard this.